packages/tools-iottwinmaker/src/lib/workspace.ts

```diff
@@ -448,19 +447,21 @@ async function deleteS3Bucket(s3BucketName: string | undefined) {
     isTruncated = listObjectVersionResp.IsTruncated;
     if (listObjectVersionResp.Versions != undefined) {
       for (const version of listObjectVersionResp.Versions) {
-        await aws().s3.deleteObject({ Bucket: s3BucketName, Key: version['Key'], VersionId: version['VersionId'] });
+        if (nonDryRun)
+          await aws().s3.deleteObject({ Bucket: s3BucketName, Key: version['Key'], VersionId: version['VersionId'] });
```

Comment:
fyi: I noticed some inconsistencies in the formatting, some places we have the `if (nonDryRun)` on single line and other places is next line with indent, let's make them consistent?

~~to clarify other comment: I think in areas like this we can have an `else` that is like: "would delete s3://<bucketname>/<key> if non-dry-run"~~ edit: actually I guess this would end up having to be plumbed through all the messages which may overcomplicate things, maybe let's just add a message at the start indicating that "all delete messages below are not actually performing delete operations"

Reply:
True, those inconsistencies are due to line length sometimes forcing an indent by ESLint. I can make it more consistent by adding braces around all if statements. I tried to be consistent by having no braces because in all cases it is a single command within the if, but perhaps that's takes away from readability.

As for second comment, I agree it might be confusing; I added a log "\nDry Run: None of the following resources will be deleted. Use --nonDryRun flag for real deletion.\n" to hopefully clarify, but open to suggestions on how to make it more clear!